Fix pre-commit config to check all model files (#637)
C2SM/icon4py: model/docs/development/ADRs/0001_model_physics_components_interface.md

**Concerns**

> - If a component depends on the tendencies produced by another component, it is unclear whether the effect of this other component

Reviewer: ```suggestion
- If a component depends on another component's tendencies, it is unclear whether the effect of the latter
```

Author: ADRs are internal technical documentation so I'd rather not modify its content, specially if written by other developers, in an unrelated PR fixing tooling configuration.

h(rho, T, ddt_T) -> ddt_T # rho, theta, ddt_theta are input only

```

All three components produce tendencies for `T`, these need to be kept separate until the state is finally updated.

> Components might depend one each other which can be made transparent through a dependency on the tendencies: in the example

Reviewer: ```suggestion
Components might depend one each other, which can be made transparent through a dependency on the tendencies: in the example
```
